x-pack/plugins/cases/public/components/create/case_owner_selection.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback, useEffect } from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiKeyPadMenu,
+  EuiKeyPadMenuItem,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+
+import { euiStyled } from '../../../../../../src/plugins/kibana_react/common';
+import { FieldHook, getFieldValidityAndErrorMessage, UseField } from '../../common/shared_imports';
+
+interface MenuSelectionProps {
+  field: FieldHook<string>;
+  isLoading: boolean;
+  availableOwners: string[];
+}
+
+interface Props {
+  availableOwners: string[];
+  isLoading: boolean;
+}
+
+const SECURITY_SOLUTION = 'securitySolution';
+const OBSERVABILITY = 'observability';
+const FIELD_NAME = 'selectedOwner';
+
+const FullWidthKeyPadMenu = euiStyled(EuiKeyPadMenu)`
+  width: 100%;
+`;
+
+const FullWidthKeyPadItem = euiStyled(EuiKeyPadMenuItem)`
+
+  width: 100%;
+`;
+
+const CaseOwnerSelectionComponent: React.FC<Props> = ({ availableOwners, isLoading }) => {
+  return (
+    <UseField
+      path={FIELD_NAME}
+      component={MenuSelection}
+      componentProps={{ availableOwners, isLoading }}
+    />
+  );
+};
+
+function MenuSelection({
+  availableOwners,
+  field,
+  isLoading = false,
+}: MenuSelectionProps): JSX.Element {
+  const { errorMessage, isInvalid } = getFieldValidityAndErrorMessage(field);
+  const radioGroupName = useGeneratedHtmlId({ prefix: 'caseOwnerRadioGroup' });
+
+  const onChange = useCallback((val: string) => field.setValue(val), [field]);
+
+  useEffect(() => {
+    if (availableOwners.length === 1) {
+      onChange(availableOwners[0]);
+    }
+  }, [availableOwners, onChange]);
+
+  return (
+    <EuiFormRow
+      data-test-subj="caseOwnerSelection"
+      fullWidth
+      isInvalid={isInvalid}
+      error={errorMessage}
+      helpText={field.helpText}
+      label={field.label}
+      labelAppend={field.labelAppend}
+    >
+      <FullWidthKeyPadMenu checkable={{ ariaLegend: 'Single case type select' }}>
+        <EuiFlexGroup>
+          <EuiFlexItem>
```

Comment:
Do we still need these Flex components using the EuiKeyPadItem with a `width: 100%`?